formats/properties/commonMain/src/kotlinx/serialization/properties/Properties.kt

```diff
@@ -101,13 +113,48 @@ public sealed class Properties(
         }
     }
 
+    private inner class InAnyMapper(
+        map: Map<String, Any>, descriptor: SerialDescriptor
+    ) : InMapper<Any>(map, descriptor) {
+        override fun structure(descriptor: SerialDescriptor): InAnyMapper =
+            InAnyMapper(map, descriptor)
+    }
+
+    private inner class InStringMapper(
+        map: Map<String, String>, descriptor: SerialDescriptor
+    ) : InMapper<String>(map, descriptor) {
+        override fun structure(descriptor: SerialDescriptor): InStringMapper =
+            InStringMapper(map, descriptor)
+
+        override fun decodeTaggedBoolean(tag: String): Boolean = decodeTaggedValue(tag).toBoolean()
+        override fun decodeTaggedByte(tag: String): Byte = decodeTaggedValue(tag).toByte()
+        override fun decodeTaggedShort(tag: String): Short = decodeTaggedValue(tag).toShort()
+        override fun decodeTaggedInt(tag: String): Int = decodeTaggedValue(tag).toInt()
+        override fun decodeTaggedLong(tag: String): Long = decodeTaggedValue(tag).toLong()
+        override fun decodeTaggedFloat(tag: String): Float = decodeTaggedValue(tag).toFloat()
+        override fun decodeTaggedDouble(tag: String): Double = decodeTaggedValue(tag).toDouble()
+        override fun decodeTaggedChar(tag: String): Char = decodeTaggedValue(tag).single()
+    }
+
     /**
      * Encodes properties from the given [value] to a map using the given [serializer].
      * `null` values are omitted from the output.
      */
     @ExperimentalSerializationApi
     public fun <T> encodeToMap(serializer: SerializationStrategy<T>, value: T): Map<String, Any> {
-        val m = OutMapper()
+        val m = OutAnyMapper()
+        m.encodeSerializableValue(serializer, value)
+        return m.map
+    }
+
+    /**
+     * Encodes properties from the given [value] to a map using the given [serializer].
+     * Converts all values to [String].
```

Comment:
I'd add that this conversion is performed with `toString` method

Reply:
Applied suggestion.